Src/Support/Google.Apis.Auth/OAuth2/ServiceAccountCredential.cs

```diff
@@ -262,10 +274,16 @@ public override async Task<bool> RequestAccessTokenAsync(CancellationToken taskC
         public override async Task<string> GetAccessTokenForRequestAsync(string authUri = null, CancellationToken cancellationToken = default)
         {
             // See: https://developers.google.com/identity/protocols/oauth2/service-account#jwt-auth
-            if (!HasExplicitScopes && authUri != null)
+            if (!HasExplicitScopes && authUri == null)
+            {
+                throw new InvalidOperationException("Invalid OAuth scope or ID token audience provided. A valid authUri and/or OAuth scope is required to proceed.");
```

Comment:
Amanda suggested that we should use the same exception type as currently gets thrown. (Probably GoogleApiException?) It's ugly, but more compatible. We may want to talk about that more. (The dependency chain may not even allow us to do that.)

Reply:
The GoogleApiException is the same type, but it requires a service name as a parameter. It throws if null, but can be empty. Not pretty, but probably better than reverting back, given that service name should be easy to figure out from the context.